NYPL-discovery/discovery-front-end
# Scc 2567 configurable base url

**What's this do?**
Supports ENV configurable `BASE_URL` and `PREVIOUS_BASE_URL`, allowing the app to listen on multiple base URLs but redirect (302) to the official base URL.

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/SCC-2567

**Do these changes have automated tests?**
No. I thought about it, but the redirect is happening in `server.js`, which doesn't currently export anything.

**How should this be QAed?**
Verify that the app behaves correctly on the following paths (and redirects properly when `PREVIOUS_BASE_URL` is configured:

/research/collections/shared-collection-catalog/
/research/collections/shared-collection-catalog/search?q=toast
/research/collections/shared-collection-catalog/bib/b13056591
/research/collections/shared-collection-catalog/hold/request/b13056591-i16219980
/research/collections/shared-collection-catalog/hold/request/b13056591-i16219980/edd
/research/collections/shared-collection-catalog/subject_headings
/research/collections/shared-collection-catalog/subject_headings/9ab1ccf8-3f70-485f-a183-45e261f1bbd7?label=Aadland%2C Florence

**Dependencies for merging? Releasing to production?**
No config is necessary for deployment to qa and production. App should behave exactly as before without setting any new variables. To test launch day redirects in QA, we'll want to set the new env vars as follows:

```
PREVIOUS_BASE_URL=/research/collections/shared-collection-catalog
BASE_URL=/research/shared-collection-catalog
```

Note that we should ask Ops to update their reverse proxy to direct requests on `/research/shared-collection-catalog` to the beanstalk apps in QA and Prod (in addition to requests on `/research/collections/shared-collection-catalog`). They should do this ASAP.

**Has the application documentation been updated for these changes?**
Updated `.env-sample`

**Did someone actually run this code to verify it works?**
I did
